### DISCUSSION TITLE
Provide a way to see full resolution of presets in the docs?

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [renovatebot/renovate#24448] [Suggest an Idea]
### Tell us more.

Coming over from https://github.com/eslint/eslint/pull/17567#discussion_r1327797853:

> > > The docs at [docs.renovatebot.com/presets-config/#configrecommended](https://docs.renovatebot.com/presets-config/#configrecommended) don't say what the explicit values are. Is that intentional? If not, would a good next step be for me to file a docs issue?
> > 
> > All the presets listed there are also documented at docs.renovatebot.com and once you drill down you can see the explicit values.
>
> I think they're trying to say that it's hard to drill down. To drill down the config:recommended preset, you'll have to have 7 tabs open for each extends item, and then try to keep it all in your head.
> 
> > Can you explain where your confusion lies? Maybe I'm not understanding what you mean by "explicit values".
>
> It sounds like they want to see all the fully expanded configuration in a single JSON codeblock. That's probably why they want to file a docs issue.

Yup, exactly. The Renovate docs make it rather inconvenient by asking us to drill down across the multiple pages. For example, the first extension name `:dependencyDashboard` isn't on https://docs.renovatebot.com/presets-config/#configrecommended - it's under https://docs.renovatebot.com/presets-default/#dependencydashboard instead. What I'm really hoping for is a way to see the exact configuration values all-at-once.

Expanding `"config:recommended"`[^1] into its constituent parts and counting the individual pages they're documented:

- `":dependencyDashboard"`[^2]:
  - `"dependencyDashboard": true`
- `":semanticPrefixFixDepsChoreOthers"`[^2]:
  - `"packageRules"`: 3 entries, including one for `"maven"` _(why would this be enabled for everyone?)_
- `":ignoreModulesAndTests"`[^2]:
  - `"ignorePaths"`: 8 entries for common example/module/test paths
- `"group:monorepos"`[^3]:
  - `"extends"`: 246 entries referring to other groups
  - `"ignoreDeps"`: an entry array _(?)_
- `"group:recommended"`[^3]:
  - `"extends"` 48 entries referring to other groups
  - `"ignoreDeps"`: an entry array _(?)_
- `"replacements:all"`[^4]:
  - `"extends"`: 25 entries referring to other replacements
  - `"ignoreDeps"`: an entry array _(?)_
- `"workarounds:all`:
  - `"extends"`: 12 entries referring to other workarounds
  - `"ignoreDeps"`: an entry array _(?)_

That's a lot of docs pages. What I'd really love is some way to download the _resolved result_. That way we can inspect the full list and see how Renovate is acting.

Is that doable at all?

Perhaps related: docs work in #16100, #21783, #22233, #23657

[^1]: https://docs.renovatebot.com/presets-config
[^2]: https://docs.renovatebot.com/presets-default
[^3]: https://docs.renovatebot.com/presets-group
[^4]: https://docs.renovatebot.com/presets-replacements
[^5]: https://docs.renovatebot.com/presets-workarounds

**Comment by rarkins** [comment]
I think a good improvement would be if we can hyperlink included presets to their respective pages. Including the _full_ config will be difficult from a UX point of view alone given how long some presets are like `group:monorepos` are.

**Reply by viceice** [reply]
linking seems to be a good compromise.

agreed that the full resolved config is too big to put on one page.

**Reply by viceice** [reply]
@HonkingGoose please open a docs issue for linking, will try to do it next week

**Reply by HonkingGoose** [reply]
> I think a good improvement would be if we can hyperlink included presets to their respective pages. 

Sounds good to me as well. See issue:

- #24456

> Including the _full_ config will be difficult from a UX point of view alone given how long some presets are like `group:monorepos` are.

We _must_ "hide" those expanded presets or we'll have endless walls of JSON configs. 😱 [Material for MkDocs has collapsible admonition blocks](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks). I'm thinking of something like this:

```markdown
<!-- prettier-ignore -->
??? note "Select me to see the fully expanded `config:recommended` preset"
    
    Insert fully expanded JSON config here
```

We should of course mock this up in a try-out PR first!

A bad side effect of this change may be that the expanded presets messes up the search. That's because some keywords are appearing a _lot_ on the page. For example: you search for `dependencyDashboard` and only see the expanded presets it's in first, before you get to the config option docs.
